components/camel-tika/src/main/java/org/apache/camel/component/tika/TikaProducer.java

```diff
@@ -59,12 +59,24 @@
     private final String encoding;
 
     public TikaProducer(TikaEndpoint endpoint) {
+        this(endpoint, true, null);
+    }
+
+    public TikaProducer(TikaEndpoint endpoint, Parser parser) {
+        this(endpoint, parser == null, parser);
+    }
+
+    private TikaProducer(TikaEndpoint endpoint, boolean autodetectParser, Parser parser) {
         super(endpoint);
         this.tikaConfiguration = endpoint.getTikaConfiguration();
         this.encoding = this.tikaConfiguration.getTikaParseOutputEncoding();
         TikaConfig config = this.tikaConfiguration.getTikaConfig();
-        this.parser = new AutoDetectParser(config);
         this.detector = config.getDetector();

```

Comment:
It seams to be a related to IDE, but looks like we don't need this for all fields.

Reply:
@dmvolod  yes, 'this.' prefixes are not required. From my point of view it helps to orientate in the code, developer immediately know, that these fields are on 'this' instance. It also helps in future, if someone adds a new parameter with the same name, it could prevent issue with initialization of the bad property.

But I have no problem with removal of 'this' on all fields (I haven't changed that) should I remove it?